workspaces/diff-engine/lib/index.js

```diff
@@ -59,9 +59,15 @@ function spawn({ specPath }) {
   diffProcess.stdout.pipe(output);
   diffProcess.stderr.pipe(error);
 
-  return { input, output, error };
+  // get a clean result promise, so we stay in control of the exact API we're exposing
```

Comment:
nice I had a feeling you'd want something like this instead of how I did it in my exploration